quarkc/test/lib/conftest.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+import json, sys
+import pexpect
+
+def pytest_collect_file(path, parent):
+    if path.ext == ".q" and path.basename.endswith("_test.q"):
+        return QuarkFile(path, parent)
+
+class QuarkFile(pytest.File):
+    def collect(self):
+        for lang in ["python", "java", "javascript", "ruby"]:
+            yield QuarkItem(self, lang)
+
+class QuarkItem(pytest.Item):
+    def __init__(self, parent, lang):
+        super(QuarkItem, self).__init__("%s:%s" % (lang, parent.fspath.basename), parent)
+        self.lang = lang
+
+    def runtest(self):
+        child = pexpect.spawn("quark", ["install", "-v"] +
+                              ["--%s" % self.lang,
+                               self.parent.fspath.strpath])
+        child.logfile = sys.stdout
+        child.expect("Done", timeout=300)
+        child.expect(pexpect.EOF)
+
+        child = pexpect.spawn(
+            "quark", ["run", "--%s" % self.lang,
+                      self.parent.fspath.strpath, "--", "--json"])
+        child.logfile = sys.stdout
+        child.expect_exact(
+            "=============================== json report ===============================",
+            timeout=300
+        )
+        child.expect(pexpect.EOF)
```

Comment:
See above regarding expecting success. This one might be okay given the code that prints `json report` is not likely to fail.